pkg/private/tar/build_tar.py

```diff
@@ -64,18 +77,14 @@ def add_file(self, f, destfile, mode=None, ids=None, names=None):
        names: (username, groupname) for the file to set ownership. `f` will be
          copied to `self.directory/destfile` in the layer.
     """
-    dest = destfile.lstrip('/')  # Remove leading slashes
-    if self.directory and self.directory != '/':
-      dest = self.directory.lstrip('/') + '/' + dest
+    dest = self.normalize_path(destfile)
     # If mode is unspecified, derive the mode from the file's mode.
     if mode is None:
       mode = 0o755 if os.access(f, os.X_OK) else 0o644
     if ids is None:
       ids = (0, 0)
     if names is None:
       names = ('', '')
-    # Note: normpath changes / to \ on windows, but we expect / style paths.
-    dest = os.path.normpath(dest).replace(os.path.sep, '/')
```

Comment:
Should we apply the same refactoring to other methods?

Reply:
The already do the same thing. But I extracted this idiom into a helper.